src/OrbitBase/include/OrbitBase/StopToken.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_STOP_TOKEN_H_
+#define ORBIT_BASE_STOP_TOKEN_H_
+
+#include <memory>
+
+#include "OrbitBase/SharedState.h"
+
+namespace orbit_base {
+
+// Forward declaration for befriending
+class StopSource;
+
+class StopToken {
+  friend StopSource;
+
+ public:
+  explicit StopToken() = default;
```

Comment:
I know the C++ library design allows default construction, but do we actually need that? I would consider not allowing that which makes it a little inconsistent in a way but also harder to mis-use.

Reply:
No, we dont need that and I will remove it. I am fine with this inconsistency, because it is more restrictive and a switch to std::stop_token should still be easy in the future. 

It means I can't create StopToken in a simple way for the unit tests. I could either remove the tests or create a StopToken via StopSource. wdyt?